packages/dd-trace/src/appsec/blocking.js

```diff
@@ -101,27 +95,31 @@ function getBlockWithContentData (req, specificType, rootSpan) {
   return { body, statusCode, headers }
 }
 
-function getBlockingData (req, specificType, rootSpan) {
-  if (blockingConfiguration?.type === 'redirect_request' && blockingConfiguration.parameters.location) {
-    return getBlockWithRedirectData(rootSpan)
+function getBlockingData (req, specificType, rootSpan, actionParameters) {
+  if (actionParameters?.location) {
+    return getBlockWithRedirectData(rootSpan, actionParameters)
   } else {
-    return getBlockWithContentData(req, specificType, rootSpan)
+    return getBlockWithContentData(req, specificType, rootSpan, actionParameters)
   }
 }
 
-function block (req, res, rootSpan, abortController, type) {
+function block (req, res, rootSpan, abortController, actionParameters) {
   if (res.headersSent) {
     log.warn('Cannot send blocking response when headers have already been sent')
     return
   }
 
-  const { body, headers, statusCode } = getBlockingData(req, type, rootSpan)
+  const { body, headers, statusCode } = getBlockingData(req, null, rootSpan, actionParameters)
 
   res.writeHead(statusCode, headers).end(body)
 
   abortController?.abort()
 }
 
+function isBlockingAction (actions) {
+  return !!(actions?.block_request || actions?.redirect_request)
```

Comment:
if you do that instead, you can do a simpler thing in other places
```suggestion
  return actions?.block_request || actions?.redirect_request
```
allows you to do:
```javascript
const actionParams = isBlockingAction(actions)
if (actionParams) {
    block(..., actionParams)
}
```
like in index.js for example, instead of repeating the `(actions.block_request || actions.redirect_request)`

Reply:
if you do it then maybe change the name to something like `getBlockingAction` or something

Reply:
`isBlockingAction` refactored to `getBlockingAction`